src/frontend/js/data/useCourseEnrollment/index.ts

```diff
@@ -20,34 +21,42 @@ const useCourseEnrollment = (resourceLink: string) => {
   const queryClient = useQueryClient();
   const EnrollmentAPI = EnrollmentApiInterface(resourceLink);
 
-  const [{ data: enrollment, isError }, queryKey] = useSessionQuery(
+  const [{ data: enrollment, isError, isLoading }, enrollmentKeys] = useSessionQuery(
     ['enrollment', resourceLink],
-    async () => EnrollmentAPI.get(user!),
+    async () => EnrollmentAPI.get(resourceLink, user!),
   );
 
-  const [{ data: isActive }] = useSessionQuery(
-    [...queryKey, 'is_active'],
+  const [{ data: isActive, refetch: refetchIsActive }] = useSessionQuery(
+    [...enrollmentKeys, 'is_active'],
     async () => EnrollmentAPI.isEnrolled(enrollment),
     {
       // Enrollment is null if it has been fetched
-      enabled: !!user && enrollment !== undefined,
+      enabled: !!user && enrollment !== undefined && !isLoading,
     },
   );
 
-  const { mutateAsync } = useSessionMutation(() => EnrollmentAPI.set(user!), {
-    mutationKey: queryKey,
-    onSuccess: () => {
-      queryClient.invalidateQueries(queryKey);
+  useEffect(() => {
+    refetchIsActive();
+  }, [enrollment]);
```

Comment:
Just style comment.

By convention, we tried to organize your component as follow :

```
const componentA = ({ firstname, lastname, ...props }) => {
  // First declare variables (state, memo, ...)
  const name = useMemo(() => `${firstname} ${lastname}`, [firstname, lastname])
  const isActive = useState(false)

  // Then methods (callback)
  const onClick = useCallback(() => ..., []);

  // Then effects
  useEffect(() => ..., []);

  // Then render
  return (<>...</>)
}
```